### PR TITLE
fix(contribs): close file in execBalancesExport

### LINKED ISSUE #3032: cmd/gnoland: execBalancesExport leaks the opened file and at no point closes it
Reading through this code https://github.com/gnolang/gno/blob/2838ad1a3c3b9795990257cd46f08fc04b2fb3a3/gno.land/cmd/gnoland/genesis_balances_export.go#L29-L78

at no point is the file that's opened per https://github.com/gnolang/gno/blob/2838ad1a3c3b9795990257cd46f08fc04b2fb3a3/gno.land/cmd/gnoland/genesis_balances_export.go#L54
ever closed

### PR DESCRIPTION
**Pull request by odeke-em**
Ensures that the opened file is not leaked and
closed after use.

Fixes #3032

### CODE REVIEW

**Comment by Gno2D2** [comment]
I'm a bot that assists the Gno Core team in maintaining this repository. My role is to ensure that contributors understand and follow our guidelines, helping to streamline the development process.

The following requirements must be fulfilled before a pull request can be merged.
Some requirement checks are automated and can be verified by the CI, while others need manual verification by a staff member.

These requirements are defined in this [configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

## Automated Checks

 🔴 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🟢 The pull request head branch must be up-to-date with its base ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch))


## Manual Checks

- [ ] The pull request description provides enough details


<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Head branch (orijtech:close-execBalancesExport-file) is up to date with base (master): behind by 0 / ahead by 1

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>The pull request description provides enough details</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 Not (🔴 Pull request author is a member of the team: core-contributors)

```
### Can be checked by
 - team core-contributors

</blockquote></details>

</blockquote></details>

</blockquote></details>

**Review by petar-dambovaliev — APPROVED** [review]

**Review by thehowl — APPROVED** [review]
it's not a problem in the real world, but it doesn't hurt.

**Comment by thehowl** [comment]
Please open any further PRs from your personal account, not your organization's account. this allows us to modify the PR and update it with master if necessary; helps us to verify the CI would not be failing on master.

**Pull request merged**
